src/hotspot/cpu/aarch64/aarch64.ad

```diff
@@ -3861,6 +3861,17 @@ encode %{
     __ mov(tmp, (address)markWord::unused_mark().value());
     __ str(tmp, Address(box, BasicLock::displaced_header_offset_in_bytes()));
 
+    __ br(Assembler::EQ, cont); // CAS success means locking succeeded
+
+    __ cmp(rscratch1, rthread);
+    __ br(Assembler::NE, cont); // Check for recursive locking
+
+    // Recursive lock case
+    __ ldr(tmp, Address(disp_hdr, ObjectMonitor::recursions_offset_in_bytes() - markWord::monitor_value));
+    __ add(tmp, tmp, 1u);
+    __ str(tmp, Address(disp_hdr, ObjectMonitor::recursions_offset_in_bytes() - markWord::monitor_value));
```

Comment:
```suggestion
    __ increment(Address(disp_hdr, ObjectMonitor::recursions_offset_in_bytes() - markWord::monitor_value));
```

Reply:
The increment macro doesn't seem to utilize the fact that 1u can be encoded as an immediate to the add instruction. So it seems to generate worse code here. I'm okay with changing to increment anyway if you prefer that.

Reply:
> The increment macro doesn't seem to utilize the fact that 1u can be encoded as an immediate to the add instruction. 

Sure it does. Try it. If it doesn't, we'll change `increment()`! 😁

Reply:
Oh yeah look at that. I disassembled it and it did the right thing. Thanks for the suggestion.